src/platform/i386/boot_comp.c

```diff
@@ -141,12 +152,16 @@ kern_boot_comp(void)
 	assert(thd_mem && tcap_mem);
         if (captbl_activate_boot(ct, BOOT_CAPTBL_SELF_CT)) assert(0);
         if (sret_activate(ct, BOOT_CAPTBL_SELF_CT, BOOT_CAPTBL_SRET)) assert(0);
-        if (pgtbl_activate(ct, BOOT_CAPTBL_SELF_CT, BOOT_CAPTBL_SELF_PT, pgtbl, 0)) assert(0);
 
 	hw_asndcap_init();
 	if (hw_activate(ct, BOOT_CAPTBL_SELF_CT, BOOT_CAPTBL_SELF_INITHW_BASE, hw_bitmap)) assert(0);
 
-	printk("\tCapability table and page-table created.\n");
+	/*
+	 * separate pgd for boot component virtual memory
+	 */
+	boot_vm_pgd = (pgtbl_t)mem_boot_alloc(1);
+	memcpy((void *)boot_vm_pgd + KERNEL_PGD_REGION_OFFSET,  (void *)(&boot_comp_pgd) + KERNEL_PGD_REGION_OFFSET, KERNEL_PGD_REGION_SIZE); 
+	if (pgtbl_activate(ct, BOOT_CAPTBL_SELF_CT, BOOT_CAPTBL_SELF_PT, (pgtbl_t)chal_va2pa(boot_vm_pgd), 0)) assert(0);
```

Comment:
I wonder if we should have a function for doing this stuff.  Is this operation done multiple times?